baseclasses/solvers/BaseSolver.py

```
@@ -118,10 +118,12 @@ def setOption(self, name, value):
                 defaultType, defaultValue = self.defaultOptions[name]
             except KeyError:
                 if name in self.deprecatedOptions:
-                    raise Error(f"Option {name} is deprecated. {self.deprecatedOptions[name]}")
+                    raise Error(f"Option {name} is deprecated. {self.deprecatedOptions[name]}") from None
                 else:
                     guess = get_close_matches(name, list(self.defaultOptions.keys()), n=1, cutoff=0.0)[0]
-                    raise Error(f"Option {name} is not a valid {self.name} option. Perhaps you meant {guess}?")
+                    raise Error(
+                        f"Option {name} is not a valid {self.name} option. Perhaps you meant {guess}?"
+                    ) from None
```

this one should actually re-raise the caught exception rather than `from None`